src/engine/objects/resolvers/visitors.rs

```diff
@@ -1529,22 +1673,30 @@ where
 
         let rel_label =
             rel_var.src().label()?.to_string() + &rel_var.label().to_title_case() + "Rel";
-        let rels = transaction
-            .update_rels::<RequestCtx>(
+        let mut rels = transaction
+            .update_rels(
                 query_fragment,
                 rel_var,
                 props,
                 props_type_name,
                 partition_key_opt,
             )
-            .await
-            .and_then(|rels| {
-                if let Some(handlers) = context.event_handlers().after_rel_update(&rel_label) {
-                    handlers.iter().try_fold(rels, |v, f| f(v))
-                } else {
-                    Ok(rels)
-                }
-            })?;
+            .await?;
+        if let Some(handlers) = context.event_handlers().after_rel_update(&rel_label) {
+            for f in handlers.iter() {
+                rels = f(
+                    rels,
+                    EventFacade::new(
+                        // TODO: where do we get rel_name?
+                        CrudOperation::UpdateRel(info.name().to_string(), "?".to_string()),
```

Comment:
The RelQueryVar struct has a function to obtain it, so it'd likely be rel_var.label().

Reply:
Fixed.